chain/indexer/src/streamer/streamer.rs

```diff
@@ -0,0 +1,228 @@
+    let mut chunks: futures::stream::FuturesUnordered<_> =
+        chunks_hashes.map(|get_chunk| fetch_single_chunk(&client, get_chunk)).collect();
+    let mut response = Vec::<views::ChunkView>::with_capacity(chunks.len());
+
+    while let Some(chunk) = chunks.next().await {
+        response.push(chunk?);
+    }
+
+    Ok(response)
+}
+
+/// Function that starts Streamer's busy loop. Every half a seconds it fetches the status
+/// compares to already fetched block height and in case it differs fetches new block of given height.
+///
+/// We have to pass `client: Addr<near_client::ClientActor>` and `view_client: Addr<near_client::ViewClientActor>`.
+pub(crate) async fn start(
+    view_client: Addr<near_client::ViewClientActor>,
+    client: Addr<near_client::ClientActor>,
+    mut blocks_sink: mpsc::Sender<BlockResponse>,
+) {
+    info!(target: INDEXER, "Starting Streamer...");
+    let mut outcomes_to_get = Vec::<types::TransactionOrReceiptId>::new();
+    let mut last_synced_block_height: types::BlockHeight = 0;
+    'main: loop {
+        time::delay_for(INTERVAL).await;
+        let status = fetch_status(&client).await;
+        if let Ok(status) = status {
+            if status.sync_info.syncing {
+                continue;
+            }
+        }
+
+        let block = if let Ok(block) = fetch_latest_block(&view_client).await {
+            block
+        } else {
+            continue;
+        };
+
+        let latest_block_height = block.header.height;
+        if last_synced_block_height == 0 {
+            last_synced_block_height = latest_block_height;
+        }
+        debug!(
+            target: INDEXER,
+            "The last synced block is #{} and the latest block is #{}",
+            last_synced_block_height,
+            latest_block_height
+        );
+        for block_height in (last_synced_block_height + 1)..=latest_block_height {
+            if let Ok(block) = fetch_block_by_height(&view_client, block_height).await {
+                let response =
+                    fetch_block_with_chunks(&view_client, block, outcomes_to_get.drain(..)).await;
+
+                if let Ok(fetch_block_response) = response {
+                    outcomes_to_get = fetch_block_response.new_outcomes_to_get;
+                    debug!(target: INDEXER, "{:#?}", &fetch_block_response.block_response);
+                    if let Err(_) = blocks_sink.send(fetch_block_response.block_response).await {
+                        info!(
+                                target: INDEXER,
+                                "Unable to send BlockResponse to listener, listener doesn't listen. terminating..."
+                            );
+                        break 'main;
+                    }
+                } else {
+                    debug!(target: INDEXER, "Missing data, skipping block #{}...", block_height);
+                }
+            }
+            last_synced_block_height = block_height;
+        }
+    }
+}
```

Comment:
This is the heart of the implementation. @bowenwang1996 @evgenykuzyakov Does this flow look reasonable to you in terms of providing a stream of blocks recorded on the chain?